File: server/router/cloudpods/network/vpc.go
package network

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type NetworkVpcRouter struct {
}

func (s *NetworkVpcRouter) InitNetworkVpcRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/network/vpc")
	vpcApi := v1.ApiGroupApp.CloudApiGroup.NetApiGroup.NetworkVpcApi

	{
		dataRouter.POST("getVpcList", vpcApi.GetVpcList)         // 获取数据列表
		dataRouter.POST("addVpc", vpcApi.AddVpc)                 // 新增
		dataRouter.POST("deleteVpc", vpcApi.DeleteVpc)           // 删除
		dataRouter.POST("deleteVpcByIds", vpcApi.DeleteVpcByIds) // 批量删除
	}
}

File: web/src/api/cloudpods/network/vpc.js
import request from '@/utils/request'
//获取
export function getVpcList(params){
    return request({
        url:'cloud/network/vpc/getVpcList',
        method:'post',
        data: params
    })
}
//创建VPC
export function addVpc(params){
    return request({
        url:'cloud/network/vpc/addVpc',
        method:'post',
        data: params
    })
}
//删除VPC
export function deleteVpc(params){
    return request({
        url:'cloud/network/vpc/deleteVpc',
        method:'post',
        data: params
    })
}
//批量删除VPC
export function deleteVpcByIds(params){
    return request({
        url:'cloud/network/vpc/deleteVpcByIds',
        method:'post',
        data: params
    })
}